tests/tests/swfs/avm1/movieclip_state_values/test.as

```diff
@@ -0,0 +1,609 @@
+			mc = createEmptyMovieClip("testMovieClip4", getNextHighestDepth());
+			mcPropsArray1 = getMcPropsArray(mc);
+
+			loadMovie("target.swf", mc);
+			break;
+		default:
+			trace("Error: Test number " + currentTest + " is invalid in this context.");
+			return;
+	}
+
+	trace("loadMovie command issued.");
+	mcPropsArray2 = getMcPropsArray(mc);
+	compareMcProps(mcPropsArray1, mcPropsArray2, true);
+
+	var frameCount = 0;
+	var initialOneFinished = false;
+	var loadingStateStarted = false;
+	var unloaded = false;
+	this.onEnterFrame = function() {
+		frameCount++;
+
+		if (!unloaded) {
+			if (mc._framesloaded == 1 && !initialOneFinished) {
+				// Sometimes in Flash, the MovieClip is still in the first state for one frame.
+				var changes = compareMcProps(mcPropsArray2, getMcPropsArray(mc), false);
+				if (changes.length != 0) {
+					// It's not the first state => We test if it's the success state.
+					initialOneFinished = true;
+				}
+			} else if (mc._framesloaded == -1) {
+				// Ruffle is (almost always) too fast to determine whether the file exists or not (and Flash often
+				// is as well); therefore this won't trace anything if the state is correct.
+				if (!loadingStateStarted) {
+					var mcPropsArrayLoadingState = getMcPropsArray(mc);
+					var differenceArray = compareMcProps(mcPropsArray2, mcPropsArrayLoadingState, false);
+					var correctDifferenceArray = [["_framesloaded", 1, -1], ["_totalframes", 1, 0],
+						["_url", "movieclip_state_values/test.swf", "movieclip_state_values/target.swf"],
+						["getBytesTotal()", 0, -1], ["getSWFVersion()", 10, -1]];
+					compareArrays(differenceArray, correctDifferenceArray, 2);
+					loadingStateStarted = true;
+				}
+			} else if (mc._framesloaded != 1) {
+				trace("Error: _framesloaded must be -1 or 1.");
+			}
+			if (mc._framesloaded == 1 && initialOneFinished) {
+				trace("Frames loaded: 1");
+				mcPropsArray3 = getMcPropsArray(mc);
+				compareMcProps(mcPropsArray2, mcPropsArray3, true);
+
+				// The MovieClip is unloaded and will enter the unloaded state on the next frame.
+				unloadMovie(mc);
+				unloaded = true;
+				trace("unloadMovie command issued.")
+				mcPropsArray4 = getMcPropsArray(mc);
+				compareMcProps(mcPropsArray3, mcPropsArray4, true);
+			}
+			initialOneFinished = true;
+		} else {
+			mcPropsArray5 = getMcPropsArray(mc);
+			compareMcProps(mcPropsArray4, mcPropsArray5, true);
+
+			this.onEnterFrame = null;
+			startNextTest(currentTest);
+			return;
+		}
+
+		if (frameCount >= 10) {
+			trace("Error: The MovieClip has not loaded.");
+			this.onEnterFrame = null;
+			startNextTest(currentTest);
+		}
+	};
+}
+
+
+
+/*
+ * The following code is unused since Ruffle tests currently don't support tests using remote URLs.
+ * It is kept here because Flash behaves differently with online SWFs and offline SWFs.
+ * If Ruffle tests start supporting remote URLs (e.g. with a local server) at some point, these tests
+ * can still be used.
```

Comment:
We probably could change the test harness to redirect remote URLs to specific files or simulate specific network conditions. Definitely out of scope for this PR...

Reply:
Yeah, Adrian17 said that it would be possible to use a local server or some kind of redirection to simulate tests with remote files.
So if this is done after this PR is merged, we can easily start using those tests as well in another PR. :)